### DISCUSSION TITLE
Clarify scope

### DISCUSSION THREAD

**Discussion by jcbhmr** [Andrew-Chen-Wang/github-wiki-action#26] [General]
I think we should clarify whether this action should include things like preprocessing links, dealing with uncommitted changes, renaming README.md ➡ Home.md etc.

Open questions

- If I want to rename README.md to Home.md in my wiki, does that belong here or do i need to manually `mv wiki/README.md wiki/Home.md` in my workflow?
- If I want to change all markdown links to strip off the .md suffix so they work on the wiki, is that in scope here, or should that be done in a preprocess step like Andrew-Chen-Wang/strip-md-suffix-markdown-links or similar?

/assign jcbhmr

_Originally posted by @jcbhmr in https://github.com/Andrew-Chen-Wang/github-wiki-action/issues/9#issuecomment-1464362090_

@Andrew-Chen-Wang are there any other scope questions that you want addressed/documented/discussed?

**Comment by jcbhmr** [comment]
**in my opinion** which I am perfectly OK with being overruled is that these are both _out of scope_ and should be encouraged to:

```yml
- uses: actions/checkout
- uses: user/strip-md-links
  with:
    path: wiki
- run: git add -A && git commit --amend --no-edit
- uses: Andrew-Chen-Wang/github-wiki-action
```

or similar

**Comment by jcbhmr** [comment]
You know I'm just now realizing how much I have just interjected myself forcefully into this project... 🤔 is this upheaval OK with you @Andrew-Chen-Wang or would you like my to pump the brakes a bit?

**Comment by Andrew-Chen-Wang** [comment]
happy to make you a co maintainer. I'll just add in a review requirement before merging for both of us. appreciate it as work has taken a toll on me which meant less OSS work!

**Comment by Andrew-Chen-Wang** [comment]
> If I want to rename README.md to Home.md in my wiki, does that belong here or do i need to manually mv wiki/README.md wiki/Home.md in my workflow?

Should be a flag default True.

> If I want to change all markdown links to strip off the .md suffix so they work on the wiki, is that in scope here, or should that be done in a preprocess step like Andrew-Chen-Wang/strip-md-suffix-markdown-links or similar?

Not sure what you mean here.

> - run: git add -A && git commit --amend --no-edit

Not fully opposed to pre-processing actions, but we want to do as much as possible inside the script and give users an option/flag unless it's not possible. Also strive to keep backwards compatibility.

**Comment by jcbhmr** [comment]
> > If I want to change all markdown links to strip off the .md suffix so they work on the wiki, is that in scope here, or should that be done in a preprocess step like Andrew-Chen-Wang/strip-md-suffix-markdown-links or similar?
> 
> Not sure what you mean here.

What I'm talking about is the problem discussed in #8 where a link like `[View our help page](./Help.md)` works **only when viewing it in the GitHub source tree** but them if you just push that file to the wiki, the `.md` suffix **won't work** because links are no longer https://github.com/Andrew-Chen-Wang/github-wiki-action/blob/master/README.md, they are https://github.com/Andrew-Chen-Wang/github-wiki-action/wiki/Test-Page-1

Note that the README.md is README.md, not just README as it would be when deployed to the wiki itself.

> If I want to change all markdown links to strip off the .md suffix so they work on the wiki, is that in scope here, or should that be done in a preprocess step like Andrew-Chen-Wang/strip-md-suffix-markdown-links or similar?

What I'm getting at here is: which of these do you want? Do **we want to offer that feature** or do we want it to be a preprocess step done by some other action?

```yml
- uses: user/strip-md-links@v1
- uses: EndBug/add-and-commit@v9
- uses: Andrew-Chen-Wang/github-wiki-action@v4
```

```yml
- uses: Andrew-Chen-Wang/github-wiki-action@v4
```

**Comment by jcbhmr** [comment]
> Not fully opposed to pre-processing actions, but we want to do as much as possible inside the script and give users an option/flag unless it's not possible. Also strive to keep backwards compatibility.

Awesome ❤ thanks for the vision clarification.

Here's some ideas of a list of features to tout on the README:
- We auto-rename README.md to Home.md
- We auto-strip .md from link suffixes
- We support auto commit of changes (default true or false?)

**Comment by jcbhmr** [comment]
> happy to make you a co maintainer. I'll just add in a review requirement before merging for both of us. appreciate it as work has taken a toll on me which meant less OSS work!

Cool! 👍 I'll try to be somewhat cautious and incremental with my changes and PRs.
